File: be/src/main/java/com/mini_ssafy_heaven/domain/enums/RoomPlayerStatus.java
package com.mini_ssafy_heaven.domain.enums;

import java.util.Arrays;
import java.util.Objects;

public enum RoomPlayerStatus {
  WAITING,
  READY,
  IN_GAME;

  public static RoomPlayerStatus get(String status) {
    return Arrays.stream(RoomPlayerStatus.values())
        .filter(playerStatus -> Objects.equals(status, playerStatus.name()))
        .findFirst()
        .orElse(WAITING);
  }

  public boolean isReady() {
    return this == READY;
  }

}

File: fe/src/constants/RoomPlayerStatus.js
export const RoomPlayerStatus = {
  WAITING: {
    value: 'WAITING',
    print: '대기 중',
  },
  READY: {
    value: 'READY',
    print: '준비완료',
  },
  IN_GAME: {
    value: 'IN_GAME',
    print: '게임 중',
  },
  isReady: (player) => {
    if (!player) {
      return false
    }

    return player.status === RoomPlayerStatus.READY.value
  },
}
